include/libtorrent/kademlia/dht_storage.hpp

```diff
@@ -64,6 +70,50 @@ namespace dht
 		void reset();
 	};
 
+	struct TORRENT_EXPORT dht_immutable_item
```

Comment:
I kind of liked that these types were hidden implementation details in the storage. I'm not sure what the best approach is though